### ISSUE TITLE
[AVM Module Issue]: Support API Server VNet Integration in `container-service/managed-cluster`

### ISSUE THREAD

**Issue by coolhome** [Azure/bicep-registry-modules#1923]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/container-service/managed-cluster

### Description

https://learn.microsoft.com/en-us/azure/aks/api-server-vnet-integration

Below is what I've been using locally and it's been working pretty well 🧑‍🚒 

```bicep
@description('Optional. Enable API Server vnet integration. Requires subscription feature flag `EnableAPIServerVnetIntegrationPreview` to be enabled during preview.')
param enableApiServerVnetIntegration bool = false

@description('Optional.  API Server subnet resource id.')
param apiServerSubnetResourceId string?
```

```bicep
    apiServerAccessProfile: {
      authorizedIPRanges: authorizedIPRanges
      disableRunCommand: disableRunCommand
      enablePrivateCluster: enablePrivateCluster
      enablePrivateClusterPublicFQDN: enablePrivateClusterPublicFQDN
      privateDNSZone: privateDNSZone
      
      // START MOD
      enableVnetIntegration: enableApiServerVnetIntegration
      subnetId: apiServerSubnetResourceId
      // END MOD
    }

```

**Comment by PixelRobots** [comment]
I think we should add this once the feature has gone GA, which will hopefully happen soon.

**Comment by AlexanderSehr** [comment]
> I think we should add this once the feature has gone GA, which will hopefully happen soon.
> 
> I also think we should add a new test deployment for this too.

I'd argee and it would also align with the repository's [specs](https://azure.github.io/Azure-Verified-Modules/specs/shared/#:~:text=Preview%20services%20and%20features%2C%20SHOULD%20NOT%20be%20promoted%20and%20exposed%2C%20unless%20they%20are%20supported%20by%20the%20respective%20PG%2C%20and%20it%E2%80%99s%20documented%20publicly.). It's technically not forbidden - but not encouraged 😉 @JPEasier, what's your take?

**Comment by AlexanderSehr** [comment]
Hey @JPEasier, could you please triage the issue 💪

**Comment by JPEasier** [comment]
Hey @coolhome,

i think we should implement your suggestion 🚀  this feature definitely makes sense.
Nevertheless this is a preview feature and we need to make sure that it is marked as such.

@AlexanderSehr  do we have conventions for AVM for preview features, for example key words in description?

**Comment by AlexanderSehr** [comment]
> Hey @coolhome,
> 
> i think we should implement your suggestion 🚀 this feature definitely makes sense. Nevertheless this is a preview feature and we need to make sure that it is marked as such.
> 
> @AlexanderSehr do we have conventions for AVM for preview features, for example key words in description?

Hey @JPEasier,
in fact we do: https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr1---category-composition---preview-services

**Comment by coolhome** [comment]
I fully agree @JPEasier this needs to be clearly marked. Thank you @AlexanderSehr for always getting the ball moving 🙌

**Comment by lopf** [comment]
AKS API Server VNet Integration has gone GA in May 2025.

Ref: https://github.com/Azure/AKS/issues/2729

**Comment by krbar** [comment]
Hi @coolhome, thank you for sharing your feedback! In the upcoming module release, we’ve revamped the parameters and added a new one called `apiServerAccessProfile`. This parameter now supports all properties provided by the resource provider, which should address the issue you encountered.
Version `0.12.0` will be available once the associated PR has been merged.

> Please note: because we've redesigned many parameters, this update will be breaking - existing deployments will need to be updated. For more details, please refer to the module changelog.

**Comment by coolhome** [comment]
💪 thank you  @krbar!
